### ISSUE TITLE
Game map shows briefly before story scenes

### ISSUE THREAD

**Issue by Wedge009** [wesnoth/wesnoth#6867]
### Game and System Information

- **Version:** 1.17.5+dev (e1f15234a40)
- **Downloaded from:** self-compiled
- **Build info:** SCons and MSVC/CMake
- **OS:** Linux/X11 and Windows

### Description of the bug

When starting a new campaign, the game map is shown briefly before the story screen loads and is displayed.

Possibly introduced with #6855. Similar, but different, to #2395, possibly related to #3421.

### Steps to reproduce the behavior

1. Start a new campaign - I observed this in HttT.
2. As the story screen loads, before it is displayed the main game interface can be seen briefly.

### Expected behavior

Ideally shouldn't see anything until the story screen loads (unless it's such a long load that it warrants a loading screen animation, but I think it'd have to be a really slow machine to take that long).

It seems similar to #2395 although that issue was to do with display of the title screen rather than the game interface.

**Comment by mesilliac** [comment]
I definitely introduced this in the draw manager commit, or at least exposed the issue. There are three potential solutions, i'm not yet sure which is the right one:
1. Implement fade-in on the entire UI as already done for the game map (it would be rather hacky for reasons related to to the GUI1 buttons in use by the theme UI).
2. Disable drawing of the game display until the story screen has finished.
3. Fix things such that there are no rendering frames in between removing the loading screen and entering the story screen

I prefer option 3 but i haven't yet looked into why there is a gap there.

**Comment by Wedge009** [comment]
No worries, just recording the issue since I hadn't seen it discussed as yet.

**Comment by mesilliac** [comment]
Hmm, well i didn't expect it would close this, but it should be fixed. Let me know if there's any flickering left.

**Comment by CelticMinstrel** [comment]
You need to avoid writing the word "fix" or "close" immediately followed by an issue number if you don't want it to auto-close the issue.

**Comment by Descacharrado** [comment]
> Hmm, well i didn't expect it would close this, but it should be fixed. Let me know if there's any flickering left.

Seems to work better now, thanks!

**Comment by Wedge009** [comment]
Yes, seems resolved, thanks. Auto-close is fine too, that's what it was designed for.

**Comment by CelticMinstrel** [comment]
Yeah, it's fine if you intended it (but in that case you need to say "fixes A and fixes B" rather than "fixes A and B".
